### PR TITLE
Small clean up on call stack section

### PR DESCRIPTION
**Pull request by miohtama**
Spelling, clarified what items go to the call stack and how any attack is executed

### CODE REVIEW

File: docs/security-considerations.rst
Comment (hrkrshnn) [inline docs/security-considerations.rst]: Maybe we should say that 63/64 rule makes call stack depth attacks impractical. What do you think @chriseth?

Comment (hrkrshnn) [inline docs/security-considerations.rst]: @miohtama Do you have an example on the effect of local function variables?

File: docs/security-considerations.rst
Comment (axic) [inline docs/security-considerations.rst]:
```suggestion
Call Stack Depth
```

File: docs/security-considerations.rst
Comment (axic) [reply docs/security-considerations.rst]: I don't think local stack items would have an effect on call stack depth.

Comment (axic) [reply docs/security-considerations.rst]: Yes, but should mention as "Starting the Spurious Dragon hard fork, the 63/64th rule ..." or whichever HF it was. Also link to the EIP.

Comment (chriseth) [reply docs/security-considerations.rst]: We should also clarify that the call stack and the expression stack are two totally unrelated things that are both limited to 1024. I guess this confusion started the PR.

Comment (hrkrshnn) [inline docs/security-considerations.rst]:
```suggestion
before they interact with your contract. Note that, since [Tangerine Whistle](https://eips.ethereum.org/EIPS/eip-608) hardfork, the [63/64 rule](https://eips.ethereum.org/EIPS/eip-150) makes call stack depth attack impractical. Also note that the call stack and the expression stack are unrelated, even though both have a size limit of 1024 stack slots.
```

File: docs/security-considerations.rst
Comment (hrkrshnn) [inline docs/security-considerations.rst]:
```suggestion
call stack size limit of 1024. In such situations, Solidity throws an exception.
```

**Comment by hrkrshnn** [comment]
I updated the doc based on suggestions. So I won't be approving this.

File: docs/security-considerations.rst
Comment (hrkrshnn) [inline docs/security-considerations.rst]:
```suggestion
before they interact with your contract. Note that, since `Tangerine Whistle <https://eips.ethereum.org/EIPS/eip-608>`_ hardfork, the `63/64 rule <https://eips.ethereum.org/EIPS/eip-150>`_ makes call stack depth attack impractical. Also note that the call stack and the expression stack are unrelated, even though both have a size limit of 1024 stack slots.
```

**Comment by miohtama** [comment]
Thank you @ @hrkrshnn! Closed the PR.

**Comment by hrkrshnn** [comment]
@miohtama Sorry I wasn't clear before. I meant that I already applied the suggestions as extra commits; so it's not fair that I approve this PR. I'd wait for someone else to approve before merging. 

Thanks for the PR!

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
Could have squashed these 4 random commits :)

**Comment by chriseth** [comment]
Argh, thought it was only one.

File: docs/security-considerations.rst
Comment (novaknole) [reply docs/security-considerations.rst]: Thanks for updating this. 

Does anyone know where the `call stack` is implemented in the source code ? I can see where each stack for each function gets created, but I can't see where it uses 1024 limit check...

Comment (chriseth) [reply docs/security-considerations.rst]: The call stack is a feature of the EVM, it is not implemented in the solidity compiler.

Comment (novaknole) [reply docs/security-considerations.rst]: Yes, I am looking at the `evm.go` or any other `go` files related to `evm`. Seems like `call stack` itself is not implemented and they only count external calls.

Comment (chriseth) [reply docs/security-considerations.rst]: Yes, this is only about external calls, that's why this paragraph starts with "External function calls".

**Pull request merged**
